erlang-ts-imenu.el

```diff
@@ -0,0 +1,83 @@
+;;; erlang-ts-imenu.el --- imenu helpers -*- lexical-binding: t; -*-
+;; %CopyrightBegin%
+;;
+;; Copyright Ericsson AB 2024-2025. All Rights Reserved.
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;     http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+;; %CopyrightEnd%
+;;
+
+;;; Commentary:
+;; Imenu functionality using treesitter
+;; Conflicts with lsp-imenu functionality if lsp is used
+;; lsp's imenu functionality can be disabled with:
+;; `(setq lsp-enable-imenu nil)`
+
+(require 'treesit)
+
+;;; Code:
+
+(defun erlang-ts-imenu-node-func-p (node)
+  "Check if NODE is the first function clause."
+  (if (equal (treesit-node-type node) "fun_decl")
+      (when-let ((prev (treesit-node-prev-sibling node))
+                 (name (erlang-ts-imenu-func-name node)))
+        (not (and (equal (treesit-node-type prev) "fun_decl")
+                  (equal (erlang-ts-imenu-func-name prev) name))))
+    nil))
+
+(defun erlang-ts-imenu-func-name (node)
+  "Fetch function name at NODE."
+  (let ((clause (treesit-node-child-by-field-name node "clause")))
+    (concat (treesit-node-text (treesit-node-child-by-field-name clause "name"))
```

Comment:
I prefer `format` over `concat` as I think it's a bit easier to read and you won't need the `number-to-string` call, but that's not a big deal.